pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -164,8 +164,9 @@ protected void doInit() {
 
       PartialUpsertHandler partialUpsertHandler = null;
       if (isPartialUpsertEnabled()) {
-        partialUpsertHandler =
-            new PartialUpsertHandler(_helixManager, _tableNameWithType, upsertConfig.getPartialUpsertStrategies());
+        partialUpsertHandler = new PartialUpsertHandler(_helixManager, _tableNameWithType, schema,
+            upsertConfig.getPartialUpsertStrategies(), upsertConfig.getDefaultPartialUpsertStrategy(),
+            tableConfig.getValidationConfig().getTimeColumnName(), upsertConfig.getComparisonColumn());
```

Comment:
This can be simplified to avoid extra handling in the method (comparisonColumn is non-null this way)
```suggestion
        String comparisonColumn = upsertConfig.getComparisonColumn();
        if (comparisonColumn == null) {
          comparisonColumn = tableConfig.getValidationConfig().getTimeColumnName();
        }
        partialUpsertHandler = new PartialUpsertHandler(_helixManager, _tableNameWithType, schema,
            upsertConfig.getPartialUpsertStrategies(), upsertConfig.getDefaultPartialUpsertStrategy(), comparisonColumn);
```